File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        	int n = height.size();
	int l = 0 , r = n - 1 ;
	int ans = 0 ;
	while (l < r){
		int temp = (r - l) * min(height[l] , height[r]);
		ans = max(temp , ans);

		if (height[l] <= height[r])
			l++;
		else
			r--;
	}

	return ans;

    }
};

File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:

        i = 0
        j = len(height)-1
        ans = 0
        while i < j:
            mn = min(height[i], height[j])
            w = j - i
            temp = mn * w
            ans = max(ans, temp)
            if height[i] <= height[j]:
                i += 1
            else:
                j -= 1
        return ans